MetaMask/metamask-mobile
# chore: bump `@metamask/mobile-provider` to `^3.0.0`

## **Description**
Catches `@metamask/providers` up several majors through `@metamask/mobile-provider`. Needed as an intermediary step before we ultimately get EIP6963 changes merged.

## **Manual testing steps**

* Open wallet (iOS)
* Go to In-app browser
* Visit https://metamask.github.io/test-dapp/
* Disconnect wallet if connected already
* Connect wallet via button on page
* Should prompt as it usually does
* Open safari
* Menu bar -> Develop -> Simulator -> Test Dapp 
* In console enter `window.ethereum.chainId`
* You should get a deprecation warning
*  In console enter `window.ethereum.chainId` again
* You should not get a second deprecation warning

## **Screenshots/Recordings**

You can verify `@metamask/providers` has been bumped if you see the deprecation errors when accessing `window.ethereum.chainId`, `window.ethereum.networkId`, and `window.ethereum.selectedAddress`  
<img width="1237" alt="Screenshot 2023-10-06 at 9 25 55 AM" src="https://github.com/MetaMask/metamask-mobile/assets/918701/5d8d1fb8-54d3-43c1-be3d-4769aaa400bc">

## **Related issues**

See: https://github.com/MetaMask/MetaMask-planning/issues/869

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
